frontend/app/routes/$lang+/_protected+/alerts+/subscribe+/confirm.tsx

```diff
@@ -0,0 +1,104 @@
+import type { ActionFunctionArgs, LoaderFunctionArgs, MetaFunction } from '@remix-run/node';
+import { json } from '@remix-run/node';
+import { useFetcher, useLoaderData, useParams } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+import pageIds from '../../../page-ids.json';
+import { Button, ButtonLink } from '~/components/buttons';
+import { ContextualAlert } from '~/components/contextual-alert';
+import { InputField } from '~/components/input-field';
+import { getInstrumentationService } from '~/services/instrumentation-service.server';
+import { featureEnabled } from '~/utils/env.server';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+import { getFixedT } from '~/utils/locale-utils.server';
+import { mergeMeta } from '~/utils/meta-utils';
+import type { RouteHandleData } from '~/utils/route-utils';
+import { getTitleMetaTags } from '~/utils/seo-utils';
+import { useUserOrigin } from '~/utils/user-origin-utils';
+
+enum ConfirmSubscriptionCode {
+  NewCode = 'new-code',
+  Submit = 'submit',
+}
+
+export const handle = {
+  breadcrumbs: [{ labelI18nKey: 'alerts:confirm.page-title' }],
+  i18nNamespaces: getTypedI18nNamespaces('alerts', 'gcweb'),
+  pageIdentifier: pageIds.protected.alerts.confirm,
+  pageTitleI18nKey: 'alerts:confirm.page-title',
+} as const satisfies RouteHandleData;
+
+export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  featureEnabled('email-alerts');
+
```

Comment:
Since this page is within the authenticated space, we need to add:

```typescript
await raoidcService.handleSessionValidation(request, session);
```

Reply:
done